Api/REST.md

```diff
@@ -1,2 +1,218 @@
+
+#### Route parameter argument
+
+Route parameter is a part of the route and should be used to access a specific resource.
+
+```c#
+	[HttpGet("{customerId}")]
+    public async Task<IActionResult> Get(Guid customerId)
+    {
+       var customer = await _customerServices.GetCustomer(customerId);
+       return Ok(customer);
+    }
+```
+
+The route to this endpoint, following the previous code, would be *`api/v1/customers/{customerId}`* where *`{customerId}`* is a *`Guid`*.
+
+#### Body of the request argument
+
+Argument can also be passed in the body of the request. By default this is JSON and is most commonly used in *`Post`* and *`Put`* methods.
+
+```c#
+	[HttpPost]
+    public async Task<IActionResult> Post([FromBody]Customer customer)
+    {
+       var customer = await _customerServices.CreateCustomer(customer);
+       return Ok(customer);
+    }
+
+	[HttpPut]
+    public async Task<IActionResult> Put(Customer customer)
+    {
+       var customer = await _customerServices.UpdateCustomer(customer);
+       return Ok(customer);
+    }
+```
+
+In *`Post`* and *`Put`* methods it is not required to put *`FromBody`* attribute as the code will work and default to it. Both of these samples would work as body arguments.
+
+#### URL parameter argument
+
+URL parameters are usually used for filtering a resources, example of pagination.
+
+```c#
+	[HttpGet]
+    public async Task<IActionResult> Get([FromQuery]Pagination pagination)
+    {
+       var paginatedCustomers = await _customerServices.GetAllCustomersPaginated(pagination);
+       return Ok(paginatedCustomers);
+    }
+```
+
+The route to this endpoint, following the previous code, would be *`api/v1/customers?pageNumber=1&pageSize=10`* where *`pageNumber`* and *`pageSize`* are public properties of *`Pagination`* class.
+
+#### Combined arguments
+
+For consistency we are usually trying to avoid combined arguments, but they are also supported and there are edge cases where they can be used. In this example, we are getting a customer information together with all orders of the customer. This can potentially be a large response and we want to provide client an option to not get order information.
+
+```c#
+	[HttpGet("{customerId}")]
+    public async Task<IActionResult> Get(Guid customerId, [FromQuery]bool includeOrders = false)
+    {
+       var customer = await _customerServices.GetCustomer(customerId, includeOrders);
+       return Ok(paginatedCustomers);
+    }
+```
+
+This endpoint now serves two routes: *`api/v1/customers/{customerId}`*, *`api/v1/customers/{customerId}?includeOrders={value}`* where *`{cusomerId}`* is GUID and *`value`* is Boolean (true/false) value.
+
+
+
+### Custom routes
+
+A custom route for action can be defined on action level. The route will inherit controllers route and add its route as a sufix.
```

Comment:
"The route will inherit controllers route and add its route as a sufix." I reword this to : if the route is relative (doesn't start with `/` or `~/`) it will be relative to Controller route - and then give some examples of relative and absolute routes